server/applicationset/applicationset.go

```diff
@@ -307,7 +307,7 @@ func (s *Server) updateAppSet(ctx context.Context, appset *v1alpha1.ApplicationS
 			return nil, err
 		}
 
-		appset, err = s.appclientset.ArgoprojV1alpha1().ApplicationSets(s.ns).Get(ctx, newAppset.Name, metav1.GetOptions{})
+		appset, err = s.appclientset.ArgoprojV1alpha1().ApplicationSets(newAppset.Namespace).Get(ctx, newAppset.Name, metav1.GetOptions{})
```

Comment:
```suggestion
		appset, err = s.appclientset.ArgoprojV1alpha1().ApplicationSets(newAppset.Namespace).Get(ctx, newAppset.Name, metav1.GetOptions{})
```
why are we using new Appset namespace here?

Reply:
That's a good question @nitishfy, it's not obvious to me either what happens here or what should happen if `newAppset.Namespace != appset.Namespace`. IMO unless there's a compelling reason to do this we should just stick with `appset.Namespace`

Reply:
Yeah, I also scratched my head a bit with this. I initially used `appset.Namespace` but then ultimately I ended up using `newAppset.Namespace` to be consistent with the fact that the [subsequent `Get`](https://github.com/argoproj/argo-cd/blob/3e6f11e08e709bb92b933106f8ca97e0ac6b3343/server/applicationset/applicationset.go#L310) is done using `newAppset.Name`.

> what should happen if newAppset.Namespace != appset.Namespace (I'm not sure if that is possible at a glance, but seems like it could be?)

Yes, I agree with this concern; the whole method is a bit weird to me. Ultimately, it takes in an existing `appset` as well as an _updated_ one. It then [overrides the spec](https://github.com/argoproj/argo-cd/blob/3e6f11e08e709bb92b933106f8ca97e0ac6b3343/server/applicationset/applicationset.go#L291) on the existing `appspec` and has some [special logic for updating the existing metadata elements](https://github.com/argoproj/argo-cd/blob/3e6f11e08e709bb92b933106f8ca97e0ac6b3343/server/applicationset/applicationset.go#L292-L299)[^1] and then updates the existing appset. The primary id (name/namespace) is assumed to be the same, which could be a fair assumption given that it's an internal function. However, the function does return the live state of the updated appset, which is why I assume the original author(s) then used `newAppset` to get the live state of whatever was supposed to be updated :grimacing: 

----

Happy to adjust anything you want to change :smile: 

[^1]: The merge logic isn't ever used outside of tests BTW - the only usage of this internal function is [here](https://github.com/argoproj/argo-cd/blob/3e6f11e08e709bb92b933106f8ca97e0ac6b3343/server/applicationset/applicationset.go#L256) where merge is set to `true`. But I figured changing this was outside the scope of this PR.

Reply:
I'm not sure if there were any special considerations taken into account with `newAppset.Namespace` vs ` appset.Namespace`. I noticed that @ishitasequeira was the original author of this - would you happen to have any input regarding this?